### PR TITLE
Add some types

### PR DESCRIPTION
**Pull request by suhr**
The situation with static typing in Python is very far from perfect, but I still hope that some types will make Mathics more maintainable.

### CODE REVIEW

File: mathics/core/definitions.py [hunk 1]
```diff
@@ -19,14 +21,14 @@
 full_names_pattern = r'(`?{0}(`{0})*)'.format(base_names_pattern)
 
 
-def get_file_time(file):
+def get_file_time(file) -> float:
     try:
         return os.stat(file).st_mtime
     except OSError:
         return 0
```
Comment (wolfv) [inline mathics/core/definitions.py]: this is an integer

Comment (suhr) [reply mathics/core/definitions.py]: But `st_mtime` is a float:

```
>>> os.stat('.zshrc').st_mtime
1548286918.3710866
```

Comment (wolfv) [reply mathics/core/definitions.py]: and mypy doesn't complain about returning the wrong number type?

Comment (suhr) [reply mathics/core/definitions.py]: Yep. I suppose integer literals can be both `int` and `float`.

Comment (wolfv) [reply mathics/core/definitions.py]: ok :)

**Comment by GarkGarcia** [comment]
Hey @suhr, thanks for rebasing the PR! I'll have a look at the CI logs, but I imagine the failed tests have nothing to do with this PR.

**Pull request merged**